controllers/mysqlcluster_controller.go

```diff
@@ -1862,6 +1861,66 @@ func (r *MySQLClusterReconciler) finalizeV1(ctx context.Context, cluster *mocov1
 	return nil
 }
 
+func (r *MySQLClusterReconciler) updateStatus(ctx context.Context, cluster *mocov1beta2.MySQLCluster, reconcileErr error) error {
+	log := crlog.FromContext(ctx)
+	orig := cluster.DeepCopy()
+
+	if cluster.Status.ReconcileInfo.Generation != cluster.Generation {
+		cluster.Status.ReconcileInfo.Generation = cluster.Generation
+		cluster.Status.ReconcileInfo.ReconcileVersion = 1
+	}
+
+	var sts appsv1.StatefulSet
+	err := r.Get(ctx, client.ObjectKey{Namespace: cluster.Namespace, Name: cluster.PrefixedName()}, &sts)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("failed to get StatefulSet %s/%s: %w", cluster.Namespace, cluster.PrefixedName(), err)
+	}
+	stsReady := metav1.ConditionFalse
+	reason := "StatefulSetNotReady"
+	message := "StatefulSet is not ready"
+	if sts.Spec.Replicas != nil && sts.Status.AvailableReplicas == *sts.Spec.Replicas && sts.Status.CurrentRevision == sts.Status.UpdateRevision && sts.Generation == sts.Status.ObservedGeneration {
+		stsReady = metav1.ConditionTrue
+		reason = "StatefulSetReady"
+		message = "StatefulSet is ready"
+	}
+	meta.SetStatusCondition(&cluster.Status.Conditions,
+		metav1.Condition{
+			Type:               mocov1beta2.ConditionStatefulSetReady,
+			Status:             stsReady,
+			ObservedGeneration: cluster.Generation,
+			Reason:             reason,
+			Message:            message,
+		},
+	)
+
+	reconcileSuccess := metav1.ConditionFalse
+	reason = "ReconcileFailed"
+	message = "reconcile failed"
+	if reconcileErr == nil {
+		reason = "ReconcileSuccess"
+		message = "reconcile successfully"
+		reconcileSuccess = metav1.ConditionTrue
+	}
```

Comment:
The order should be unified.

```suggestion
	reconcileSuccess := metav1.ConditionFalse
	reason = "ReconcileFailed"
	message = "reconcile failed"
	if reconcileErr == nil {
		reconcileSuccess = metav1.ConditionTrue
		reason = "ReconcileSuccess"
		message = "reconcile successfully"
	}
```

Reply:
fixed